astroid/protocols.py

```diff
@@ -397,24 +397,25 @@ def arguments_assigned_stmts(
     context: InferenceContext | None = None,
     assign_path: list[int] | None = None,
 ) -> Any:
-    if context.callcontext:
+    try:
+        node_name = node.name  # type: ignore[union-attr]
+    except AttributeError:
+        node_name = None
```

Comment:
See comment above. It's a bit non-sensical I think to add coverage and I wouldn't even know how to do it other than directly calling the function itself.

We could add a pragma? But perhaps leaving as is is also good enough?

Reply:
```suggestion
        # The typing of node includes types that do not have a name attribute. mypy was
        #  rightfully complaining about this as it could create an uncaught AttributeError
        # We have resolved this by looking up the attribute within this try ... except.
        # https://github.com/PyCQA/astroid/pull/1644#discussion_r901545816
        node_name = None  # pragma: no cover
```

Reply:
```suggestion
        # Added to handle edge cases where node.name is not defined.
        # See https://github.com/PyCQA/astroid/pull/1644
        node_name = None  # pragma: no cover
```

Reply:
```suggestion
        # Added to handle edge cases where node.name is not defined.
        # https://github.com/PyCQA/astroid/pull/1644#discussion_r901545816
        node_name = None  # pragma: no cover
```
Sure ! But let's link to the main comment of the PR.